sdk/typescript/runtime/main.go

```diff
@@ -223,18 +177,117 @@ func (t *TypescriptSdk) Base(runtime SupportedTSRuntime) (*Container, error) {
 			WithExec([]string{"apk", "add", "ca-certificates"}).
 			WithEnvVariable("NODE_OPTIONS", "--use-openssl-ca").
 			WithMountedCache("/root/.npm", dag.CacheVolume(fmt.Sprintf("mod-npm-cache-%s", nodeVersion))).
+			WithMountedCache("/usr/local/share/.cache", dag.CacheVolume(fmt.Sprintf("mod-yarn-cache-%s", nodeVersion))).
 			WithExec([]string{"npm", "install", "-g", "tsx"}), nil
 	default:
 		return nil, fmt.Errorf("unknown runtime: %s", runtime)
 	}
 }
 
+// setupModule initialiaze the user's module.
+//
+// If the user's module has a package.json file, it will run the
+// __tsconfig.updator.ts script in order to add dagger to the tsconfig path so
+// the editor can give type hints and auto completion.
+// Otherwise, it will copy the template config files into the user's module directory.
+//
+// If there's no src directory or no typescript files in it, it will create one
+// and copy the template index.ts file in it.
+func (t *TypescriptSdk) setupModule(ctx context.Context, ctr *Container, runtime SupportedTSRuntime, name string) (*Container, error) {
+	moduleFiles, err := ctr.Directory(".").Entries(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not list rootfs entries: %w", err)
+	}
+
+	packageJSONExist := slices.Contains(moduleFiles, "package.json")
+
+	// If there's a package.json, run the tsconfig updator script.
+	// else, copy the template config files.
+	if packageJSONExist {
+		executor := "tsx"
+
+		if runtime == Bun {
+			executor = "bun"
+		}
+
+		ctr = ctr.WithExec([]string{executor, "/opt/module/bin/__tsconfig.updator.ts"})
```

Comment:
Why not put this logic in Go since it's faster?

Reply:
That's a good point, however I think I should take care of that in a follow up